src/tests/posix/memccpy_test.c

```diff
@@ -0,0 +1,174 @@
+/**
+ * @file
+ * @brief Tests for memccpy function
+ * 
+ * @author Giannis Zapantis  (github name: tech-gian)
+ * @date 18/3/2020
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <embox/test.h>
+
+
+EMBOX_TEST_SUITE("memccpy");
+
+
+/* TEST CASES for testing memccpy
+ * on strings
+ */
+
+
+TEST_CASE("empty static allocated string with character off the string") {
+    
+    char original[] = "To be copied";
+    char tested[128];
+
+    void* test = memccpy(tested, original, 'w', strlen(original)+1);
+    test_assert_zero(strcmp(tested, original));
+    test_assert_null(test);
+}
+
+
+TEST_CASE("non-empty static allocated string with character off the string") {
+
+    char original[] = "To be copied";
+    char tested[] = "Before copy";
+
+    void* test = memccpy(tested, original, 'w', strlen(original)+1);
+    test_assert_zero(strcmp(tested, original)); 
+    test_assert_null(test);
+}
+
+
+TEST_CASE("non-empty static allocated string with character on the string") {
+
+    char original[] = "To be copied";
+    char tested[] = "Before copy";
+
+    void* test = memccpy(tested, original, 'c', strlen(original)+1);
+    test_assert_zero(strcmp(tested, "To be  copy"));
+    test_assert_not_null(test);
+}
+
+
+TEST_CASE("empty static allocated string with character on the string") {
+
+    char original[] = "To be copied";
+    char tested[128];
+
+    void* test = memccpy(tested, original, 'c', strlen(original)+1);
+    test_assert_zero(strcmp(tested, "To be "));
+    test_assert_not_null(test);
+}
+
+
+TEST_CASE("dynamic allocated string with character off the string") {
+
+    char original[] = "To be copied";
+    char* tested = malloc(128*sizeof(char));

```

Comment:
Where do you free memory

Reply:
I add free for all the pointers, that I called with malloc.